sycl/doc/SYCLDriverforRFC.md

```diff
@@ -0,0 +1,53 @@
+# Adjust the compiler driver to support offloading for SYCL kernels
+
+Supporting the SYCL programming model requires a number of adjustments to the
+compiler driver. These changes involve introducing enabling command line
+options, adjusting the compilation tool chains to provide device specific
+compilations as well as corresponding host compilations. The behaviors are
+centered around compilation of source files, consumption of objects and
+libraries that contain SYCL kernels will be done by the `clang-linker-wrapper`.
+
+## Command line options for enabling.
+
+A number of command line options will be introduced to enable the offloading
+compilation for SYCL.
+* -fsycl
+  * Enables the device and host compilations against a given source file.
+* -fsycl-targets=\<target\>
+  * Compiles the device based on the given target value. 
+* -fsycl-device-only
+  * Enables the ability to create only device code.
+* -fsycl-host-compiler
+  * Provide the ability to use a 3rd party compiler to perform the host
+    compilation step.
+
```

Comment:
Right - `-fsycl-is-device` is used when compiling for device, and `-fsycl-is-host` for the host side of the same compilation toolchain.  `-sycl-std=` is used to provide the std version so that should stick around too.